connections/sim/hw_sim.py

```diff
@@ -132,6 +132,27 @@ def read(self):
     def fire(self):
         self._on_level = self.DISCONNECTED
 
+class Clock:
+    def __init__(self) -> None:
+        self._time_us = 0
+
+    def add_time(self, delta_us: float) -> None:
+        """
+        :param delta_us: number of microseconds to add to the clock.
+        """
+        self._time_us += delta_us
```

Comment:
```suggestion
        self._time_us += int(delta_us)
```
Lets always cast the delta to an int to ensure that `self._time_us` doesn't somehow become a float.

Reply:
Fixed it missed that one.